pkg/apis/resource/validation/validation.go

```diff
@@ -1192,3 +1195,22 @@ func validateLabelValue(value string, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+func validateDeviceBindingParameters(bindingConditions, bindingFailureConditions []string, bindingTimeoutSeconds *int64, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateSlice(bindingConditions, resource.BindingConditionsMaxSize,
+		metav1validation.ValidateLabelName, fldPath.Child("bindingConditions"))...)
+
+	allErrs = append(allErrs, validateSlice(bindingFailureConditions, resource.BindingFailureConditionsMaxSize,
+		metav1validation.ValidateLabelName, fldPath.Child("bindingFailureConditions"))...)
+
+	if bindingTimeoutSeconds != nil && *bindingTimeoutSeconds <= 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("bindingTimeoutSeconds"), bindingTimeoutSeconds, "must be greater than zero"))
+	}
+
+	if bindingConditions == nil && bindingFailureConditions != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("bindingConditions"), bindingConditions, "bindingConditions must be set if bindingFailureConditions are set"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath.Child("bindingConditions"), bindingConditions, "bindingConditions are required to use bindingFailureConditions"))
```

Slight wording change to explain why this restriction exists.